### PR TITLE
doc: update docs and versions for NCS 2.0.1

### PR DESCRIPTION
**Pull request by greg-fer**
Updated structure and release notes.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
@edmont , can you also check the Thread known issues, just to be sure all need 2.0.1?

File: doc/nrf/known_issues.rst [hunk 1]
```diff
@@ -371,21 +372,21 @@ Missing :file:`CMakeLists.txt`
 Thread
```
Comment (edmont) [inline doc/nrf/known_issues.rst:372]: Yes, the below Thread issues apply to NCS 2.0.1 as well.

File: doc/nrf/releases/release-notes-2.0.1.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
+The feature is made available in its current state though the design and interfaces can change between release tags.
+The feature will also be labelled with "EXPERIMENTAL" in Kconfig files to indicate this status.
+Build warnings will be generated to indicate when features labelled EXPERIMENTAL are included in builds unless the Kconfig option :kconfig:option:`CONFIG_WARN_EXPERIMENTAL` is disabled.
+
+Highlights
+**********
+
+* Updated the :ref:`nrfxlib:ot_libs` in nrfxlib and made minor changes related to Thread.
```
Comment (edmont) [inline doc/nrf/releases/release-notes-2.0.1.rst:21]: Might be worth mentioning that the changes affect/fix Thread 1.2 Certification?

**Review by edmont — APPROVED** [review]

File: doc/nrf/releases/release-notes-2.0.1.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/releases/release-notes-2.0.1.rst:21]: Yes, let's wait for Krzysztof's highlights first.

Comment (greg-fer) [reply doc/nrf/releases/release-notes-2.0.1.rst]: Done.

**Review by anangl — APPROVED** [review]

**Review by doublemis1 — APPROVED** [review]

File: doc/nrf/releases/release-notes-2.0.1.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+.. contents::
+   :local:
+   :depth: 2
+
+|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products in the nRF52, nRF53, and nRF91 Series.
+The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and re-distributed with the SDK.
+
+Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
+The feature is made available in its current state though the design and interfaces can change between release tags.
+The feature will also be labelled with "EXPERIMENTAL" in Kconfig files to indicate this status.
```
Comment (Avalei) [inline doc/nrf/releases/release-notes-2.0.1.rst:15]:
```suggestion
The feature will also be labeled with "EXPERIMENTAL" in Kconfig files to indicate this status.
```

File: doc/nrf/releases/release-notes-2.0.1.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+Build warnings will be generated to indicate when features labelled EXPERIMENTAL are included in builds unless the Kconfig option :kconfig:option:`CONFIG_WARN_EXPERIMENTAL` is disabled.
```
Comment (Avalei) [inline doc/nrf/releases/release-notes-2.0.1.rst:16]:
```suggestion
Build warnings will be generated to indicate when features labeled EXPERIMENTAL are included in builds unless the Kconfig option :kconfig:option:`CONFIG_WARN_EXPERIMENTAL` is disabled.
```

File: doc/nrf/releases/release-notes-2.0.1.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+To use this release, check out the tag in the manifest repository and run ``west update``.
+See :ref:`cloning_the_repositories` for more information.
+
+For information on the included repositories and revisions, see `Repositories and revisions for v2.0.1`_.
+
+IDE and tool support
+********************
+
+`nRF Connect extension for Visual Studio Code <nRF Connect for Visual Studio Code_>`_ is the only officially supported IDE for |NCS| v2.0.1.
+SEGGER Embedded Studio Nordic Edition is no longer tested and recommended for new projects.
```
Comment (Avalei) [inline doc/nrf/releases/release-notes-2.0.1.rst:42]:
```suggestion
SEGGER Embedded Studio Nordic Edition is no longer tested or recommended for new projects.
```

**Review by Avalei — CHANGES_REQUESTED** [review]

**Comment by greg-fer** [comment]
@Avalei , edits applied.

**Review by shantha-14 — APPROVED** [review]

**Pull request merged**